### ISSUE TITLE
overlayfs failure

### ISSUE THREAD

**Issue by AkihiroSuda** [weber-software/diuid#7]
```console
$ docker run ---rm  weberlars/diuid@sha256:d884fc60a64aa0e2cac87fac731199aff7fa21bd9b4b9202da353f0e090ca06c docker run --rm debian:9.9 bash -c "apt-get update && apt-get install -y gcc"
[ ok ] Starting OpenBSD Secure Shell server: sshd.
For better performance, consider mounting a tmpfs on /umlshm like this: `docker run --tmpfs /umlshm:rw,nosuid,nodev,exec,size=8g`
waiting for dockerd .......
Unable to find image 'debian:9.9' locally
9.9: Pulling from library/debian
6f2f362378c5: Pull complete
Digest: sha256:118cf8f3557e1ea766c02f36f05f6ac3e63628427ea8965fb861be904ec35a6f
Status: Downloaded newer image for debian:9.9
Ign:2 http://cdn-fastly.deb.debian.org/debian stretch InRelease
Get:3 http://cdn-fastly.deb.debian.org/debian stretch-updates InRelease [91.0 kB]
...
Get:24 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 libc-dev-bin amd64 2.24-11+deb9u4 [259 kB]
Get:25 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 libc6-dev amd64 2.24-11+deb9u4 [2364 kB]
Get:26 http://cdn-fastly.deb.debian.org/debian stretch/main amd64 manpages-dev all 4.10-2 [2145 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 29.5 MB in 4s (6598 kB/s)
Selecting previously unselected package manpages.
(Reading database ... 6499 files and directories currently installed.)
Preparing to unpack .../00-manpages_4.10-2_all.deb ...
Unpacking manpages (4.10-2) ...
dpkg: error processing archive /tmp/apt-dpkg-install-nRslku/00-manpages_4.10-2_all.deb (--unpack):
 unable to install new version of './usr/share/doc/manpages': Invalid cross-device link
dpkg-deb: error: subprocess paste was killed by signal (Broken pipe)
Selecting previously unselected package binutils.
Preparing to unpack .../01-binutils_2.28-5_amd64.deb ...
Unpacking binutils (2.28-5) ...
dpkg: error processing archive /tmp/apt-dpkg-install-nRslku/01-binutils_2.28-5_amd64.deb (--unpack):
 unable to install new version of './usr/lib/compat-ld': Invalid cross-device link
dpkg-deb: error: subprocess paste was killed by signal (Broken pipe)
...
Unpacking manpages-dev (4.10-2) ...
dpkg: error processing archive /tmp/apt-dpkg-install-nRslku/25-manpages-dev_4.10-2_all.deb (--unpack):
 unable to install new version of './usr/share/man/man2': Invalid cross-device link
dpkg-deb: error: subprocess paste was killed by signal (Broken pipe)
Errors were encountered while processing:
 /tmp/apt-dpkg-install-nRslku/00-manpages_4.10-2_all.deb
 /tmp/apt-dpkg-install-nRslku/01-binutils_2.28-5_amd64.deb
 /tmp/apt-dpkg-install-nRslku/02-libgmp10_2%3a6.1.2+dfsg-1_amd64.deb
 /tmp/apt-dpkg-install-nRslku/03-libisl15_0.18-1_amd64.deb
 /tmp/apt-dpkg-install-nRslku/04-libmpfr4_3.1.5-1_amd64.deb
 /tmp/apt-dpkg-install-nRslku/05-libmpc3_1.0.3-1+b2_amd64.deb
 /tmp/apt-dpkg-install-nRslku/07-cpp_4%3a6.3.0-4_amd64.deb
 /tmp/apt-dpkg-install-nRslku/19-libgcc-6-dev_6.3.0-18+deb9u1_amd64.deb
 /tmp/apt-dpkg-install-nRslku/20-gcc-6_6.3.0-18+deb9u1_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

source: 2c585abb7db84f70aed2869978aab276d0a60bb3

**Comment by AkihiroSuda** [comment]
should be fixed after upgrading to Docker 19.03 (will be released in a few days)
https://github.com/moby/moby/pull/37993

**Comment by AkihiroSuda** [comment]
Attaching an ext4 disk with `-e DISK=$((4 << 30))` seems solve the issue.

Maybe the upstream overlayfs has an issue when the lowerdir is tmpfs
